### ISSUE TITLE
[RFF] JSON output logging

### ISSUE THREAD

**Issue by aress31** [byt3bl33d3r/CrackMapExec#484]
Right now parsing the result output from CME is not really intuitive, need to come up with crafty bash one-liners, etc. Why not offering JSON output so that tester could easily extract relevant information, might it be, the host, user, password, shares or payload?

**Comment by ict-one-nl** [comment]
I really really would this option as well. It does exactly what we need, but I need to enrich the data with our CMDB. To do that I need to parse the data somehow. JSON output would be much appreciated.

**Comment by SmashRocket** [comment]
Would be a great feature +1

**Comment by mpgn** [comment]
The sponsors version has a log feature where everything is send to a file.

**Comment by Marshall-Hallenbeck** [comment]
@aress31 @ict-one-nl @SmashRocket @Senne-s @daanwa Do you mean the entire output (so instead of the normal output, it's all JSON), or just an option to have normal STDOUT prints and JSON to a file, or a way to dump CMEDB information as JSON? Could you give some examples and I can see what I can put together to fit the needs.

**Comment by daanwa** [comment]
I want the results of the tool to be incorporated into our tooling. However, the current output shown on STDOUT is somewhat cumbersome to parse. So in my case it would be most handy to have a command line flag that outputs the results in JSON to a file.

Preferably I would not have to use the cmedb cli and I could just provide the JSON output flag to the cme app.

**Comment by Marshall-Hallenbeck** [comment]
@daanwa Thanks for that information. It might be pretty hard to do, but I can look into it. Can you provide an example of how you would expect the JSON structure to look like with several hosts & modules?

**Comment by ict-one-nl** [comment]
Yup, I'd like to parse the output with something like a python script to write to opensearch and enrich the info with contents of our CMDB :)

**Comment by ict-one-nl** [comment]
That's up to my scripting skills (to ingest data into our SIEM and generate reports). But to do anything useful I need to interact with the data and for that I'd need parsable output and json works fine for that :)

**Comment by Marshall-Hallenbeck** [comment]
@ict-one-nl What sort of structure would you expect as output? Can you provide an example?

**Comment by ict-one-nl** [comment]
Hmm, doesn't really matter, I can image a header like object for the general stuff and a list of objects for each line outputted by CME containing all the details for that line?

**Comment by mpgn** [comment]
but what's the point ? I don't understand

**Comment by daanwa** [comment]
Exactly. It is mainly aimed at integrating with some of our other tooling or for collecting additional authentication information about certain data points. 

@mpgn Perhaps there is already a different way that can do this and I'm just not aware of it? Currently I'm parsing the sqlite databases but I've seen that not all commands/modules populate the db's.

**Comment by ict-one-nl** [comment]
Using the SQLite db quite cumbersome for what's needed in this case. Just a different formatter for the data that's already there would suffice :)

**Comment by ict-one-nl** [comment]
Okay, did it the sqlite way for now:

sqlite3 -json /root/.cme/default/smb.db 'select * from shares inner join computers on computers.hostname = shares.computerid  where read = true and name NOT LIKE "%print%"' > /root/.cme/output.json

**Comment by mpgn** [comment]
So, my 2 cents, if we add a json option it should be used in combinaison with `--log file` option.

`cme <proto> <ip> -u <user> -p <pass> --log filename --json`

And the output in the log file 'filename' will be in json. I know @NeffIsBack already work on this, I think this is the best way to do it.

**Comment by daanwa** [comment]
Yeah that would work for me!
